dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/imports/HarvestedJsonParser.java

```diff
@@ -330,100 +280,133 @@ private Date parseTime(String str) throws ParseException {
                 }
             }
-
-            vals.add(parentField);
-        }
-        if (!vocabExceptions.isEmpty()) {
-            throw new CompoundVocabularyException("Invalid controlled vocabulary in compound field ", vocabExceptions, vals);
+            if (parentField.getDatasetFieldsChildren().size() > 0) {
+                vals.add(parentField);
+            }
         }
         return vals;
     }
 
-    private List<DatasetField> parsePrimitiveValue(DatasetFieldType dft, JsonObject json) throws JsonParseException {
+    private List<DatasetField> parsePrimitiveValue(DatasetFieldType dft, JsonObject json) {
 
         List<DatasetField> vals = new LinkedList<>();
-        if (dft.isAllowMultiples() && json.get("value").getValueType().equals(JsonValue.ValueType.ARRAY)) {
-
-            for (JsonString val : json.getJsonArray("value").getValuesAs(JsonString.class)) {
-                DatasetField datasetFieldValue = new DatasetField();
-                datasetFieldValue.setDisplayOrder(vals.size() - 1);
-                datasetFieldValue.setFieldValue(val.getString().trim());
-                datasetFieldValue.setDatasetFieldType(dft);
-                vals.add(datasetFieldValue);
-            }
 
-        } else {
-            try {
-                json.getString("value");
-            } catch (ClassCastException cce) {
-                throw new JsonParseException("Invalid value submitted for " + dft.getName() + ". It should be a single value.");
+        try {
+            if (dft.isAllowMultiples()) {
+                try {
+                    vals.addAll(parsePrimitiveMultiValues(dft, json));
+                } catch (ClassCastException | NullPointerException exception) {
+                    vals.add(parsePrimitiveSingleValue(dft, json));
+                }
+            } else {
+                try {
+                    vals.add(parsePrimitiveSingleValue(dft, json));
+                } catch (ClassCastException | NullPointerException exception) {
+                    List<DatasetField> multiValues = parsePrimitiveMultiValues(dft, json);
+                    if (multiValues.size() > 0) {
+                        vals.add(multiValues.get(0));
+                    }
+                }
             }
+        } catch (ClassCastException | NullPointerException ex) {
+            return Collections.emptyList();
+        }
+
+        return vals;
+    }
+
+    private List<DatasetField> parsePrimitiveMultiValues(DatasetFieldType dft, JsonObject json)
+            throws ClassCastException, NullPointerException {
+        List<DatasetField> vals = new LinkedList<>();
+        for (JsonString val : json.getJsonArray("value").getValuesAs(JsonString.class)) {
             DatasetField datasetFieldValue = new DatasetField();
-            datasetFieldValue.setFieldValue(json.getString("value", "").trim());
+            datasetFieldValue.setDisplayOrder(vals.size() - 1);
```

Comment:
I think that `displayOrder` will be filled either way in further processing and it doesn't really matters but in current solution we will be starting from index `-1`